### PR TITLE
Rename in adls failing even after operation being sucessful

### PR DESCRIPTION
**Pull request by vibhansa-msft**
- if file/dir does not exists in local cache then even when rename is successful, on cli we return failure "Input/Output error"

### CODE REVIEW

File: blobfuse/src/DataLakeBfsClient.cpp [hunk 1]
```diff
@@ -325,24 +325,30 @@ std::vector<std::string> DataLakeBfsClient::Rename(const std::string /*sourcePat
...
 
-    if(rename_ret != 0)
-    {
-        syslog(LOG_ERR, "Failure to rename source file %s in the local cache.  Errno = %d.\n", sourcePath.c_str(), errno);
+        if(rename_ret != 0) {
+            syslog(LOG_ERR, "Failure to rename source file %s in the local cache.  Errno = %d.\n", sourcePath.c_str(), errno);
+        }
+        
+        file_paths_to_remove.push_back(sourcePath);
+        errno = 0;
```
Comment (NaraVen) [inline blobfuse/src/DataLakeBfsClient.cpp:350]: 1)
rename_cached_file should not return a non zero errno if rename_ret =0. Please fix line add errno=0 after 473 of rename_cached_file 
2) this line doesn't make sense, to simply reset the errno, if the file is in local cache and rename_cached_file could not rename there is something wrong the error has to bubble up.
3) same with file_paths_to_remove, if the local rename has failed when it is in cache what will we achieve by adding the oldpath to the list to be removed from cache .

**Review by NaraVen — CHANGES_REQUESTED** [review]

File: blobfuse/src/DataLakeBfsClient.cpp (hunk 1 above)
Comment (vibhansa-msft) [reply blobfuse/src/DataLakeBfsClient.cpp:350]: - Code modified to take care of point 1,2.
- For point 3 : the files are renamed in container but local rename has failed. If we keep the files in local cache directory listing will show both of them.

File: blobfuse/src/DataLakeBfsClient.cpp [hunk 2]
```diff
@@ -325,24 +325,27 @@ std::vector<std::string> DataLakeBfsClient::Rename(const std::string /*sourcePat
...
+        syslog(LOG_ERR, "Failure to rename source file %s in container.  Errno = %d.\n", sourcePath.c_str(), errno);
+    } else {
+        std::string srcMntPathString = prepend_mnt_path_string(sourcePath);
+        std::string dstMntPathString = prepend_mnt_path_string(destinationPath);
+
+        long int rename_ret = rename_cached_file(srcMntPathString.c_str(), dstMntPathString.c_str());
+        if(rename_ret != 0) {
+            syslog(LOG_ERR, "Failure to rename source file %s in the local cache.  Errno = %d.\n", sourcePath.c_str(), errno);
+        } 
+        file_paths_to_remove.push_back(sourcePath);
```
Comment (NaraVen) [inline blobfuse/src/DataLakeBfsClient.cpp:347]: sourcePath should not be added to the file_paths_to_remove if the file does not exist in cache. So the return from rename_cached_file should return a tuple, the second value of the pair should indicate if the file was in cache or not.

**Review by NaraVen — CHANGES_REQUESTED** [review]

File: blobfuse/src/DataLakeBfsClient.cpp (hunk 2 above)
Comment (vibhansa-msft) [reply blobfuse/src/DataLakeBfsClient.cpp]: Added a check to push to vector only when local cache was cleared.

**Review by NaraVen — APPROVED** [review]

**Pull request merged**